core/src/test/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacadeTest.java

```diff
@@ -0,0 +1,482 @@
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.mockito.stubbing.Answer;
+
+import static com.novoda.notils.logger.simple.Log.setShowLogs;
+import static org.fest.assertions.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+
+public class AndroidMediaPlayerFacadeTest {
+
+    private static final int ANY_DURATION = 12000;
+    private static final int ANY_POSITION = 60;
+    private static final int ANY_WIDTH = 100;
+    private static final int ANY_HEIGHT = 50;
+    private static final int ANY_ERROR_WHAT = -1;
+    private static final int ANY_ERROR_EXTRA = 404;
+    private static final int TEN_PERCENT = 10;
+    private static final int MINUS_ONE = -1;
+    private static final int ZERO = 0;
+    private static final int TEN_SECONDS_IN_MILLIS = 10000;
+
+    private static final boolean SCREEN_ON = true;
+    private static final boolean IS_IN_PLAYBACK_STATE = true;
+    private static final boolean IS_NOT_IN_PLAYBACK_STATE = false;
+    private static final boolean IS_PLAYING = true;
+    private static final boolean IS_NOT_PLAYING = false;
+
+    private static final Map<String, String> NO_HEADERS = null;
+    private static final Uri ANY_URI = mock(Uri.class);
+    private static final PlayerAudioTrack PLAYER_AUDIO_TRACK = new PlayerAudioTrack(0, 0, "id", "english", ".mp4", 1, 120);
+    private static final List<PlayerAudioTrack> AUDIO_TRACKS = Collections.singletonList(PLAYER_AUDIO_TRACK);
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private Context context;
+    @Mock
+    private AndroidMediaPlayerAudioTrackSelector trackSelector;
+    @Mock
+    private PlaybackStateChecker playbackStateChecker;
+    @Mock
+    private MediaPlayer mediaPlayer;
+    @Mock
+    private AudioManager audioManager;
+    @Mock
+    private SurfaceHolderRequester surfaceHolderRequester;
+    @Mock
+    private SurfaceHolder surfaceHolder;
+    @Mock
+    private MediaPlayer.OnPreparedListener preparedListener;
+    @Mock
+    private MediaPlayer.OnVideoSizeChangedListener videoSizeChangedListener;
+    @Mock
+    private MediaPlayer.OnErrorListener errorListener;
+    @Mock
+    private MediaPlayer.OnCompletionListener completionListener;
+
+    private AndroidMediaPlayerFacade facade;
+
+    @Before
+    public void setUp() {
+        setShowLogs(false);
+
+        facade = new AndroidMediaPlayerFacade(
+                context,
+                audioManager,
+                trackSelector,
+                playbackStateChecker
+        ) {
+            @Override
+            protected MediaPlayer createMediaPlayer() {
```

Comment:
An option could be to have a `MediaPlayerCreator` or something like that, right? that would be the dependency of the class and lazily.

Can be done on a later PR